packages/connect-node-conformance/src/conformance.ts

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021-2023 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as os from "node:os";
+import {
+  readFileSync,
+  createReadStream,
+  writeFileSync,
+  existsSync,
+  chmodSync,
+  mkdirSync,
+} from "node:fs";
+import { join as joinPath } from "node:path";
+import { unzipSync } from "fflate";
+import * as tar from "tar-stream";
+import { pipeline } from "node:stream/promises";
+import { execFileSync } from "node:child_process";
+import { createGunzip } from "node:zlib";
```

Comment:
This cargo-culted from other release configs, where "tgz" is the norm, and "zip" is provided for Windows since that's the only format it supports out-of-the-box (users would have to download a separate tool or install wsl for a "tgz" file).

Does this have to be downloaded via JS code? I guess I was thinking this is easiest with a simple shell script that uses curl or wget and tar. The suffix of the file is designed to work with the output of uname.

FWIW, this appears to also be how we do releases of the CLI: https://github.com/bufbuild/buf/releases/tag/v1.28.1

Reply:
I'll leave it up to you guys on how to handle it, but I agree with @jhump that a shell script pulling down the binary would be the easiest. I understand the idiomatic thing, but since this is more of an ancillary process and not involved in the actual logic of Connect-ES maybe a shell script is ok? At least maybe pulling down the tarball and un-tarring it?

Reply:
Shell scripts in Connect-ES, Protobuf-ES, and other repositories targeting TS and JS are not okay. They are a barrier for most users.

Reply:
> `fflate` is a great package, but can we use `node:zlib` here, since we already use it?

I don't think we can use zlib to handle zip archives. But I removed the zlib usage to avoid using two different packages for decompression.

Reply:
Right, I [remember now](https://github.com/timostamm/protobuf-ts/blob/6cdc5cb27c2fef0faf6a0d79fe1ede27b7335a54/packages/protoc/util.js#L328) :joy: